backend/src/main/java/de/jankahle/capstone/service/RecipeService.java

```diff
@@ -31,7 +31,9 @@ public RecipeService(RecipeMongoDB recipeMongoDB, ImageReaderService imageReader
 
     public Recipe saveRecipe(RecipeDto recipeDto) {
         Recipe recipe = recipeDto.toRecipe();
-        recipe.setId(IdUtility.generateId());
+        if (recipe.getId().isEmpty()) {
+            recipe.setId(IdUtility.generateId());
```

Comment:
Why not include this in your models `toRecipe()` method?

Reply:
The toRecipe() should never generate values by itself only copy them into a new Recipe object.